content/en/blog/_posts/2021-10-05-nsa-cisa-hardening.md

```diff
@@ -0,0 +1,413 @@
+Beware of the possibility of malicious
+shell [code execution](https://banzaicloud.com/blog/kubeconfig-security/) in a
+`kubeconfig` file. Once attackers gain access to the cluster, they can steal ssh
+keys/secrets or more.
+
+### Secrets
+
+[External secrets providers](https://github.com/external-secrets/kubernetes-external-secrets)
+and [csi-secrets-store](https://github.com/kubernetes-sigs/secrets-store-csi-driver)
+can also be used as alternatives to Kubernetes secrets.
+
+## Log Auditing
+
+The report stresses monitoring and alerting based on logs. The key points
+include logging at the host level, application level, and on the cloud. When
+running kubernetes in production, it's important to understand who's
+responsible, and who's accountable, for each layer of logging.
+
+### Kubernetes API auditing
+
+One area that deserves more focus is what exactly should alert or be logged. The
+document outlines a sample policy in *Appendix L: Audit Policy* that logs all
+RequestResponse's including metadata and request / response bodies. While
+helpful for a demo, it may not be practical for production.
+
+Each organization needs to evaluate their
+own [threat model](https://en.wikipedia.org/wiki/Threat_model) and build an
+audit policy that complements or helps troubleshooting incident response. Think
+about how someone would attack your organization and what audit trail could
+identify it. Review more advanced options for tuning audit logs in
+the [official audit logging documentation](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#audit-policy)
+. It's crucial to tune your audit logs to only include events that meet your
+threat model.
+
+<!–– IF https://github.com/kubernetes-sigs/kind/pull/2431 merges... Audit
+logging configurations can also be tested with
+kind ([instructions](https://deploy-preview-2431--k8s-kind.netlify.app/docs/user/auditing/))
+. ––>
+
+### Streaming logs and auditing
+
+Logging is important for threat and anomaly detection. As the document outlines,
+it's a best practice to scan and alert on logs as close to real time as possible
+and to protect logs from tampering if a compromise occurs. It's important to
+reflect on the various levels of logging and identify the critical areas such as
+API endpoints.
+
+Kubernetes API audit logging can stream to a webhook and there's an example in *
+Appendix N: Webhook configuration*. Using a webhook could be a method that
+stores logs off cluster and/or centralizes all audit logs. Once logs are
+centrally managed, look to enable alerting based on critical events. Also ensure
+you understand what the baseline is for normal activities.
+
+### Alert identification
+
+While the guide stressed the importance of notifications, there is not a blanket
+event list to alert from. The alerting requirements vary based on your own
+requirements and threat model. A few examples could be:
+
+- Changes to SecurityContext of a pod
```

Comment:
```suggestion
- Changes to the `securityContext` of a Pod
```
(writing Pod in title case because here it refers to Pod as the API, not the equivalent thing that runs on a node).